src/system.h

```diff
@@ -103,19 +110,68 @@ class SystemCommon
         atom_style = "atomic";
 
         mass = t_mass();
-        domain_x = domain_y = domain_z = 0.0;
-        sub_domain_x = sub_domain_y = sub_domain_z = 0.0;
-        domain_lo_x = domain_lo_y = domain_lo_z = 0.0;
-        domain_hi_x = domain_hi_y = domain_hi_z = 0.0;
-        sub_domain_hi_x = sub_domain_hi_y = sub_domain_hi_z = 0.0;
-        sub_domain_lo_x = sub_domain_lo_y = sub_domain_lo_z = 0.0;
+
+        global_mesh_x = global_mesh_y = global_mesh_z = 0.0;
+        local_mesh_lo_x = local_mesh_lo_y = local_mesh_lo_z = 0.0;
+        local_mesh_hi_x = local_mesh_hi_y = local_mesh_hi_z = 0.0;
+        ghost_mesh_lo_x = ghost_mesh_lo_y = ghost_mesh_lo_z = 0.0;
+        ghost_mesh_hi_x = ghost_mesh_hi_y = ghost_mesh_hi_z = 0.0;
+        local_mesh_x = local_mesh_y = local_mesh_z = 0.0;
+
         mvv2e = boltz = dt = 0.0;
 
         mass = t_mass( "System::mass", ntypes );
     }
 
     ~SystemCommon() {}
 
+    void create_domain( std::array<double, 3> low_corner,
+                        std::array<double, 3> high_corner )
+    {
+        // Create the MPI partitions.
+        Cajita::UniformDimPartitioner partitioner;
```

Comment:
Are most MD simulations usually in a domain shaped like a uniform cube? If so this partitioner should always be OK.

Reply:
LAMMPS has both options, but I think it makes sense to stick with this and explore the non-uniform partitioning later